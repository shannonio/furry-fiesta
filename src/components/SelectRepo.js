import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';

import logo from '../images/octocat.png';
import './SelectRepo.scss';

const SelectRepo = ({ repos, fetchRepos, history }) => {
  const [selectedRepo, setSelectedRepo] = useState({});

  if (!repos) {
    fetchRepos();
    return null;
  }

  if (!selectedRepo.id) {
    setSelectedRepo(repos[0]);
  }

  const selectRepo = (repo) => {
    setSelectedRepo(repo);
    history.push(`/repo/${repo.owner.login}/${repo.name}`);
  }

  const isOptionSelected = (opt) => opt.id === selectedRepo.id;

  const selectValues = repos.map((repo, idx) => (
    <div className="select-box__value" key={idx}>
      <input className="select-box__input"
             type="radio"
             id={idx}
             value={repo.id}
             name={repo.name}
             readOnly={true}
             checked={isOptionSelected(repo)}
      />
      <p className="select-box__input-text">{repo.name}</p>
    </div>
  ));

  const selectOptions = repos.map((repo, idx) => (
    <li onClick={() => selectRepo(repo)} key={idx}>
      <label className="select-box__option" htmlFor={idx} aria-hidden="aria-hidden">
        { repo.name }
      </label>
    </li>
  ));

  return (
    <div className="SelectRepo">
      <div className="SelectRepo__select-box-container">
        <div className="select-box">
          <div className="select-box__current" tabIndex="1">
            { selectValues }
            <img className="select-box__icon" src="http://cdn.onlinewebfonts.com/svg/img_295694.svg" alt="Arrow Icon" aria-hidden="true"/>
          </div>
          <ul className="select-box__list">
            { selectOptions }
          </ul>
        </div>
      </div>
    </div>
  );
}


export default withRouter(SelectRepo);
