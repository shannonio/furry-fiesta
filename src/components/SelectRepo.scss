.SelectRepo {
  &__select-box-container {
    padding: 3rem;
  }
}
