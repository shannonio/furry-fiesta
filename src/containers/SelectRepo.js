import { connect } from 'react-redux';
import fetchRepos from '../actions/repos';
import { getRepos } from '../reducers/repos';
import SelectRepo from '../components/SelectRepo'

const mapStateToProps = (state, ownProps) => {
  return {
    repos: getRepos(state)
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRepos: () => fetchRepos()(dispatch)
  }
};

const SelectRepoContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(SelectRepo);

export default SelectRepoContainer
