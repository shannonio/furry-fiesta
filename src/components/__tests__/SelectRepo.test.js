import React from 'react';
import ReactDOM from 'react-dom';
import { render, getByDisplayValue} from '@testing-library/react'

import SelectRepo from '../SelectRepo';

describe('SelectRepo', () => {
  const mockFetch = jest.fn();
  const repos = [
    {
      id: 123,
      name: 'foo'
    }
  ];

  const createComponent = props => (
    render(
      <SelectRepo {...props} />
    )
  )


  it('renders without crashing', () => {
    const props = {
      repos,
      fetchRepos: mockFetch
    };

    const component = createComponent(props);

    expect(component).toMatchSnapshot();
  });

  it('has a selected option on init', () => {
    const props = {
      repos,
      fetchRepos: mockFetch
    };

    const { container } = createComponent(props);

    const selectedRepo = getByDisplayValue(container, "123");
    expect(selectedRepo.checked).toBe(true);
  });
});
