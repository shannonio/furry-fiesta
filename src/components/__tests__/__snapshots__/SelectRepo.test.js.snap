// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRepo renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="SelectRepo"
      >
        <div
          class="SelectRepo__select-box-container"
        >
          <div
            class="select-box"
          >
            <div
              class="select-box__current"
              tabindex="1"
            >
              <div
                class="select-box__value"
              >
                <input
                  checked=""
                  class="select-box__input"
                  id="0"
                  name="foo"
                  readonly=""
                  type="radio"
                  value="123"
                />
                <p
                  class="select-box__input-text"
                >
                  foo
                </p>
              </div>
              <img
                alt="Arrow Icon"
                aria-hidden="true"
                class="select-box__icon"
                src="http://cdn.onlinewebfonts.com/svg/img_295694.svg"
              />
            </div>
            <ul
              class="select-box__list"
            >
              <li>
                <label
                  aria-hidden="aria-hidden"
                  class="select-box__option"
                  for="0"
                >
                  foo
                </label>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="SelectRepo"
    >
      <div
        class="SelectRepo__select-box-container"
      >
        <div
          class="select-box"
        >
          <div
            class="select-box__current"
            tabindex="1"
          >
            <div
              class="select-box__value"
            >
              <input
                checked=""
                class="select-box__input"
                id="0"
                name="foo"
                readonly=""
                type="radio"
                value="123"
              />
              <p
                class="select-box__input-text"
              >
                foo
              </p>
            </div>
            <img
              alt="Arrow Icon"
              aria-hidden="true"
              class="select-box__icon"
              src="http://cdn.onlinewebfonts.com/svg/img_295694.svg"
            />
          </div>
          <ul
            class="select-box__list"
          >
            <li>
              <label
                aria-hidden="aria-hidden"
                class="select-box__option"
                for="0"
              >
                foo
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
